### PR TITLE
Replace the Joiner in EnvironmentId and Cluster from Underscore to sl…

### PR DESCRIPTION
**Pull request by gongmax**
#### Summary
1. Replace the Joiner in Class EnvironmentId, Cluster and AttributeDDBConverter from underscore to slash.
2. Re-organize class ApiModelMapper's import, so now all the imported classes that have the same name in both data-service and data-service-model are from com.amazonaws.blox.dataservice.model.

#### Testing
./gradlew build
./gradlew check

New tests cover the changes: no

#### Licensing

This contribution is under the terms of the Apache 2.0 License: Yes

### CODE REVIEW

File: data-service/src/main/java/com/amazonaws/blox/dataservice/model/Cluster.java [hunk 1]
```diff
@@ -29,12 +29,12 @@
 // required for mapstruct
 @NoArgsConstructor
 public class Cluster {
-  private static final String UNDERSCORE = "_";
+  private static final String JOINER = "_";
```
Comment (wbingli) [inline data-service/src/main/java/com/amazonaws/blox/dataservice/model/Cluster.java]: Still underscore?

Comment (wbingli) [reply data-service/src/main/java/com/amazonaws/blox/dataservice/model/Cluster.java]: maybe to use `DELIMITER` as the var name?

File: data-service/src/main/java/com/amazonaws/blox/dataservice/model/EnvironmentId.java [hunk 2]
```diff
@@ -29,30 +29,30 @@
 @NoArgsConstructor
 public class EnvironmentId {
 
-  private static final String UNDERSCORE = "_";
+  private static final String JOINER = "/";
```
Comment (wbingli) [inline data-service/src/main/java/com/amazonaws/blox/dataservice/model/EnvironmentId.java]: Actually AWS ARN doesn't forbid to use "/" in the resource even slash is used for ARN format like `arn:partition:service:region:account-id:resourcetype/resource`, you may see some services use '/' in their resource ID.

But inside ECS,   we don't allow '/' in the name, e..g in the cluster name, it says "The name of your cluster. If you do not specify a name for your cluster, you create a cluster named default. Up to 255 letters (uppercase and lowercase), numbers, hyphens, and underscores are allowed."

I think use slash is good idea so far.  The UNDERSCORE is definitely buggy as cluster name allowed to contain underscore.

File: data-service/src/main/java/com/amazonaws/blox/dataservice/repository/model/AttributeDDBConverter.java [hunk 3]
```diff
@@ -20,21 +20,18 @@
 
 public class AttributeDDBConverter implements DynamoDBTypeConverter<String, Attribute> {
 
-  private static final String UNDERSCORE = "_";
+  private static final String JOINER = "_";
```
Comment (wbingli) [inline data-service/src/main/java/com/amazonaws/blox/dataservice/repository/model/AttributeDDBConverter.java]: still underscore?

File: data-service/src/main/java/com/amazonaws/blox/dataservice/model/EnvironmentId.java (hunk 2 above)
Comment (xjhe) [reply data-service/src/main/java/com/amazonaws/blox/dataservice/model/EnvironmentId.java]: Should we add this message in Blox front end as well, so people know that '/' is not allowed in names? 
We can do this in another sim...not necessarily in this PR.

Comment (wbingli) [reply data-service/src/main/java/com/amazonaws/blox/dataservice/model/EnvironmentId.java]: I don't think we allow it. We can follow the same name requirement as cluster, like "Up to 255 letters (uppercase and lowercase), numbers, hyphens, and underscores are allowed.". In this way,  it basically doesn't allow slash.

Comment (xjhe) [reply data-service/src/main/java/com/amazonaws/blox/dataservice/model/EnvironmentId.java]: That's true. My suggestion is that we should output this similar message to the user when they try to use names with slash :-D

File: data-service/src/main/java/com/amazonaws/blox/dataservice/model/Cluster.java (hunk 1 above)
Comment (gongmax) [reply data-service/src/main/java/com/amazonaws/blox/dataservice/model/Cluster.java]: Yeah, that's the word... Fixed.

File: data-service/src/main/java/com/amazonaws/blox/dataservice/model/EnvironmentId.java (hunk 2 above)
Comment (gongmax) [reply data-service/src/main/java/com/amazonaws/blox/dataservice/model/EnvironmentId.java]: We are getting these cluster names only from ECS, so there is no need for Blox to validate the cluster name. But yeah, we definitely need to add validation and return some message to indicate the invalid parameters when customer call createEnvironment in FE. Will check and address it in future PR.

**Comment by wbingli** [comment]
Looks build failed, style issue.

**Comment by gongmax** [comment]
Fix the style issue

**Review by wbingli — APPROVED** [review]
LGTM

**Pull request merged**